scheduler/scheduler.cpp

```diff
@@ -1766,16 +1770,19 @@ static int prepare_broadcast_reply(char* buf, const char* netname)
     }
 }
 
-static void broadcast_scheduler_version()
+static void broadcast_scheduler_version(const char* netname)
 {
-    const int schedbuflen = 4 + sizeof(uint64_t);
+    char length_netname = sizeof(netname);
+    const int schedbuflen = 5 + sizeof(uint64_t) + length_netname;
     char buf[ schedbuflen ];
     buf[0] = 'I';
     buf[1] = 'C';
     buf[2] = 'E';
     buf[3] = PROTOCOL_VERSION;
     uint64_t tmp_time = starttime;
     memcpy(buf + 4, &tmp_time, sizeof(uint64_t));
+    buf[4 + sizeof(uint64_t)] = length_netname;
+    memcpy(buf + 5 + sizeof(uint64_t), netname, length_netname);
     DiscoverSched::broadcastData(scheduler_port, buf, sizeof(buf));
```

Comment:
We need to send two messages for now, one with the old format (no netname), one with the new one.  Otherwise we break mixed networks.